### DISCUSSION TITLE
Feature: gmove command could chose the group automatically when there's only 2

### DISCUSSION THREAD

**Discussion by fstamour** [stumpwm/stumpwm#1118] [Ideas]
When we use the command `gmove` to move a window to another group and there is only 2 groups, the logical choice is "the other group" (i.e. not the current group). 

We could make the command automatically choose the other group in this case.

There are surely other commands that could benefit from this, `gmove-and-follow` at least.

What do you think?

EDIT:

IIUC, only this needs to be modified: https://github.com/stumpwm/stumpwm/blob/189a730f73266999d5b8f992faf939c11d7815ad/command.lisp#L482-L487

**Comment by sdilts** [comment]
I think it's a good idea! However, we have a strong policy against changing stuff like this, as it could break other users' scripts. A new command would need to be added. Maybe it should be called `gmove-smart`?

Another way to do this could be implementing `gmove-other`, which would use the same heuristic to find which group to pick as `gother`. With only two groups, it would have similar behavior to what you described.

**Comment by fstamour** [comment]
I think a "configuration" would be better. A bit like the mouse-focus-policy.

**Comment by dmb2** [comment]
I like this idea too but I agree with @sdilts.  Really we could float this on stumpwm-devel and see what the greater community thinks about a breaking change... This is a pretty obvious new behavior.

Thinking out loud: I wonder how hard it would be to add something like emacs's defadvice which would enable users to advise/reconfigure behaviors like this ad-hoc.

**Comment by fstamour** [comment]
Just to reiterate: it doesn't need to be a breaking change if it's hidden behind a configuration.

**Comment by dmb2** [comment]
If you submit a PR that has this behavior *and* makes it configurable, with the default behavior being the current behavior I'm happy to merge it.
